worker/task/process_event_delivery.go

```diff
@@ -245,14 +245,33 @@ func ProcessEventDelivery(endpointRepo datastore.EndpointRepository, eventDelive
 			requestLogger.Errorf("%s", eventDelivery.UID)
 			done = false
 
-			eventDelivery.Status = datastore.RetryEventStatus
-
-			nextTime := time.Now().Add(delayDuration)
-			eventDelivery.Metadata.NextSendTime = nextTime
-			attempts := eventDelivery.Metadata.NumTrials + 1
-
-			log.FromContext(ctx).Errorf("%s next retry time is %s (strategy = %s, delay = %d, attempts = %d/%d)\n", eventDelivery.UID,
-				nextTime.Format(time.ANSIC), eventDelivery.Metadata.Strategy, eventDelivery.Metadata.IntervalSeconds, attempts, eventDelivery.Metadata.RetryLimit)
+			// For at-most-once delivery, only retry on network failures
+			if eventDelivery.DeliveryMode == datastore.AtMostOnceDeliveryMode {
+				if retryableForAtMostOnceDeliveryMode(resp.StatusCode) {
+					// Network error - retry
+					eventDelivery.Status = datastore.RetryEventStatus
+					nextTime := time.Now().Add(delayDuration)
+					eventDelivery.Metadata.NextSendTime = nextTime
+					attempts := eventDelivery.Metadata.NumTrials + 1
+
+					log.FromContext(ctx).Errorf("%s next retry time is %s (strategy = %s, delay = %d, attempts = %d/%d)\n", eventDelivery.UID,
+						nextTime.Format(time.ANSIC), eventDelivery.Metadata.Strategy, eventDelivery.Metadata.IntervalSeconds, attempts, eventDelivery.Metadata.RetryLimit)
+				} else {
+					// Got a response (even if it's an error status code) - mark as failed
+					eventDelivery.Status = datastore.FailureEventStatus
+					eventDelivery.Description = fmt.Sprintf("Endpoint returned status code %d", statusCode)
+					done = true
```

Comment:
Our API requires the status code needs to be recorded. An array in the event delivery would be ideal (for appending redelivery attempts).

Reply:
I thought so but I wasn't able to tell if it was being set as expected (here or in tests).